src/MySqlConnector/Protocol/Serialization/ProtocolUtility.cs

```diff
@@ -89,12 +88,6 @@ ValueTask<ArraySegment<byte>> AddContinuation(ValueTask<Packet> readPacketTask_,
 
 		private static bool HasReadPayload(ArraySegmentHolder<byte> previousPayloads, Packet packet, ProtocolErrorBehavior protocolErrorBehavior, out ValueTask<ArraySegment<byte>> result)
 		{
-			if (packet == null && protocolErrorBehavior == ProtocolErrorBehavior.Ignore)
-			{
-				result = default(ValueTask<ArraySegment<byte>>);
-				return true;
-			}
```

Comment:
`packet` will never be `null` now, because I made it a struct. OTOH, I'm not sure if it was ever `null` before, because it's dereferenced later in the method without a `null` check. (If it was possible that `default(Packet)` could come through, then `packet.Contents.Array` might throw a `NullReferenceException` later in the method.)